apps/passport-server/src/apis/fetch.ts

```diff
@@ -13,7 +15,20 @@ export function instrumentedFetch(
     span?.setAttribute("method", init?.method ?? "GET");
 
     try {
-      const result = await fetch(input, init);
+      const result = await execWithRetry(
+        () => {
+          // eslint-disable-next-line no-restricted-globals
+          return fetch(input, init);
+        },
+        (e) => {
+          const errorMessage = getErrorMessage(e);
+          // some endpoints sometimes close our connection from their end
+          // which is a case we want to retry.
+          return errorMessage.includes("other side closed");
```

Comment:
i don't totally understand this - is checking if the error message contains this string a good enough metric for whether the other side closes their connection?

Reply:
It's good enough for me - not sure what the best practices are. my logic is that these are errors coming directly from the fetch api, which i doubt would be overlapping (i.e. no false positives/negatives w/ the string comparison method)